lib/sandbox/sandbox.go

```diff
@@ -460,6 +460,29 @@ func (s *Sandbox) NetNsCreate() error {
 	return nil
 }
 
+// UserNsPath returns the path to the user namespace of the sandbox.
+// If the sandbox uses the same namespace as the caller, nil is returned
+func (s *Sandbox) UserNsPath() (*string, error) {
+	hostNS, err := os.Readlink("/proc/self/ns/user")
+	if err != nil {
+		return nil, err
+	}
+	if s.infraContainer != nil {
```

Comment:
This should be checked before reading the link to /proc/self/ns/user, since this is not requieed if this is not an infrastructure container.